src/core/responses.ts

```
@@ -7,13 +7,13 @@ export function addBadRequest(queryParams?: unknown, requestBody?: unknown) {
   return { description: "Bad Request" } as ResponseObject;
 }
 
-export function bundleResponses(collection: Record<string, ResponseDefinition>) {
-  return Object.entries(collection).reduce((result, [key, response]) => {
+export function bundleResponses<Defs extends Record<string, unknown>>(collection: ResponseCollection<Defs>) {
+  return Object.entries(collection).reduce((result, [key, response]: [string, ResponseDefinition<unknown>]) => {
     return {
       ...result,
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)